### DISCUSSION TITLE
Allow to add a comment for why dashboard approval is needed

### DISCUSSION THREAD

**Discussion by renchap** [renovatebot/renovate#23796] [Suggest an Idea]
### Tell us more.

`dependencyDashboardApproval` is a very nice feature, but I would like to be able to specify a comment displayed in the dashboard above the dependency, so I can add a note of why approval is needed as there is a specific reason for those packages to be requiring approval.

Some examples:
- on a Ruby Dockerfile base image: "This needs to be bumped when the `.ruby-version` file is bumped to a new major" (same for Node & other languages)
- on a `postgresql` image in a `docker-compose.yml` file: "Any major upgrades here will require users to manually upgrade their postgres data to a new version, this is a complex process and needs some warnings"

These use cases come from the [Mastodon repository](https://github.com/mastodon/mastodon).

**Comment by HonkingGoose** [comment]
Renovate can put the `description` from _presets_ into the onboarding PR body text: [^1]

> The description field can be used inside any configuration object to add a human-readable description of the object's config purpose. Descriptions fields embedded within presets are also collated as part of the onboarding description.

So there's already _some_ support for "take user's `description` and put it in the PR body text". But there are some differences as well:

1. Dependency Dashboard is an issue, not a PR.
2. Where to put the text on the Dependency Dashboard?
3. Some platforms limit the length of the Issue text, so we'll have to decide on truncation priorities (if there's not enough space, drop the `description` in favor of showing more update checkboxes, for example).
4. Where should the user put the `description` in the `renovate.json` file, so Renovate can pick it up, and display it?
5. Users may want to have `description` in their Renovate config, but don't want the text on the dashboard?
6. Maybe use a different name for the "show text on Dependency Dashboard" configuration option? For example: `dependencyDashboardDescription`.

I'll let the maintainers decide _if_ and _how_ to create this feature. 😉

**Edit**, I found some related issues:

- https://github.com/renovatebot/renovate/issues/17749
- https://github.com/renovatebot/renovate/issues/12414

[^1]: [Renovate docs, `description` config option](https://docs.renovatebot.com/configuration-options/#description)

**Reply by HonkingGoose** [reply]
By the way, I found `dependencyDashboardTitle` and `dependencyDashboardHeader` and `dependencyDashboardFooter`, so we already have _some_ way of putting a comment into the Dependency Dashboard. But it's a generic thing, it's not customized to show _why_ you're wanting approval for a specific update first... 😄

**Comment by viceice** [comment]
a good candidate for extending

https://docs.renovatebot.com/configuration-options/#customizedashboard

@PhilipAbed 😉

**Reply by PhilipAbed** [reply]
yeah, i was asking my self some questions about this, then i saw honking goose's message and he asked the same questions, there's a lot of unknowns.

**Reply by HonkingGoose** [reply]
@rarkins can you review the whole discussion and tell us what you think? 😉

**Reply by rarkins** [reply]
Can someone mock up how the markdown would look/work within the dashboard? Should be possible to edit source of an existing dashboard, edit it, then post to a gist

**Reply by renchap** [reply]
Here are 2 proposals, from [Mastodon's Dashboard](https://github.com/mastodon/mastodon/issues/25299)

https://gist.github.com/renchap/fcf42c89ebc16464eb1ffc138de79f75

**Reply by PhilipAbed** [reply]
i thought about it as: Add an action next to each branch PR title, like #1

in the #2 you show adding action sub sections for a specific branch updates

so maybe something like this for input? ( using the `customizeDashboard`)
`customizeDashboard = {approvalReasons: {<reason>, [...packageNames]}}`

do we want this only for `pending approval` section? or do we want it for more sections in the future? 

it's easier to implement using the #1 where we can add actions per package, its more generic.
``customizeDashboard = {packageActions: {<action>, [...packageNames]}}``

**Comment by rarkins** [comment]
> Here are 2 proposals, from [Mastodon's Dashboard](https://github.com/mastodon/mastodon/issues/25299)

I like the idea of option 1 (putting it on the same line) although maybe surrounding it with `( )`

**Reply by renchap** [reply]
It really depends if you have specific notes for each entry, in that case option is it better, or if you have multiple packages that are using the same note (in our case, we have multiple packages that require a Webpack 5 update, so it would repeat the node for all of them, and wont show them grouped)

**Reply by rarkins** [reply]
So let's say that in theory a branch could have multiple upgrades and each upgrade a different approval reason. Should we combine and de-dupe them, or pick one?
